mesonbuild/compilers/c.py

```diff
@@ -73,7 +73,10 @@ def __init__(self, exelist: T.List[str], version: str, for_machine: MachineChoic
         CLikeCompiler.__init__(self, exe_wrapper)
 
     def get_no_stdinc_args(self) -> T.List[str]:
-        return ['-nostdinc']
+        if get_id() == 'msvc':
```

Comment:
In python you need to explicitly reference self, unlike C++ and Ruby, so this should be:
```python
if self.get_id() == 'msvc':
   return ['/X']
else:
    return ['-nostdinc']
```

However, that said, I'd prefer to implement the correct value in the subclasses rather than having an if here. Since I don't have access to a number of the compilers we support to test with, the safest way to do this would be to just add an override method into the `VisualStudioLikeCCompilerMixin` class (I've checked that ClangCL, and I can check the IntelCL compiler later), do the same for C++.

Reply:
To be evern more clear, we've had a long layering violation, really the `-nostdinc` and `-nostdlib` should be in the `GnuLikeCompiler` layer, but I don't know which other compilers might be affected by that change.